typescript/lib/networks/network_context.ts

```diff
@@ -0,0 +1,541 @@
+
+    /*
+    Creates a new NetworkContext object with the given parameters.
+    */
+    constructor(
+            client: ApiContainerServiceClient,
+            enclaveDataVolMountpoint: string) {
+                this.client = client;
+                this.enclaveDataVolMountpoint = enclaveDataVolMountpoint;
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async loadLambda(
+            lambdaId: LambdaID,
+            image: string,
+            serializedParams: string): Promise<Result<LambdaContext, Error>> {
+        const args: LoadLambdaArgs = newLoadLambdaArgs(lambdaId, image, serializedParams);
+        
+        const promiseLoadLambda: Promise<ResultAsync<any, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.loadLambda(args, (error: grpc.ServiceError, response: any) => {
+                if (error) {
+                    resolve(errAsync(error))
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultLoadLambda: Result<any, Error> = await promiseLoadLambda;
+        if (!resultLoadLambda.isOk()) {
+            return err(resultLoadLambda.error);
+        }
+
+        const moduleCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return ok(moduleCtx);
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async getLambdaContext(lambdaId: LambdaID): Promise<Result<LambdaContext, Error>> {
+        const args: GetLambdaInfoArgs = newLambdaInfoArgs(lambdaId);
+        
+        const promiseGetLambdaInfo: Promise<ResultAsync<GetLambdaInfoResponse, Error>> = new Promise((resolve, _unusedReject) => {
+            this.client.getLambdaInfo(args, (error: grpc.ServiceError, response: GetLambdaInfoResponse) => {
+                if (error) {
+                    resolve(errAsync(error));
+                } else {
+                    resolve(okAsync(response));
+                }
+            })
+        });
+        const resultGetLambdaInfo: Result<GetLambdaInfoResponse, Error> = await promiseGetLambdaInfo;
+        if (!resultGetLambdaInfo.isOk()) {
+            return err(resultGetLambdaInfo.error);
+        }
+
+        const lambdaCtx: LambdaContext = new LambdaContext(this.client, lambdaId);
+        return ok(lambdaCtx);
+    }
+
+    // Docs available at https://docs.kurtosistech.com/kurtosis-libs/lib-documentation
+    public async registerStaticFiles(staticFileFilepaths: Map<StaticFileID, string>): Promise<Result<null, Error>> {
+        const strSet: Map<string, boolean> = new Map();
+        for (let [staticFileId, srcAbsFilepath] of staticFileFilepaths.entries()) {
+            
+            // Sanity-check that the source filepath exists
+            const promiseSrcAbsFilepath: Promise<fs.Stats> = fsPromises.stat(srcAbsFilepath);
+            promiseSrcAbsFilepath.catch(() => { 
```

Comment:
This is very close, but not quite what you want - remember that the callback inside the `catch` function will be executed _when an error is thrown in the promise_, which will be at some point in the future. This `return err(new Error(.....` will a) only `return` from inside the callback which b) will be executed at some point in the future if/when an error is thrown.

I.e., we're creating `promiseSrcAbsFilepath` which has the deferred catch logic... but then we're not doing anything with it so it will run, and if an error is thrown then it will run your error callback..... but the value that your error callback returns will never be used. You _could_ put an `await` here so that we block until we get the value back (which is what we want), but reading the `await` docs say that if the Promise is rejected then `await` will throw an exception with the Promise rejection value (and exceptions should be exceptional, so we don't want that).

In thinking on this, I _think_ what you'll need to do here is wrap `fsPromises` with our Promise-catching infrastructure just like you do above (and in which case, it might actually make sense to just use the regular `fs.stat` inside a `try/catch`)